### ISSUE TITLE
Missing translations (found errors)

### ISSUE THREAD

**Issue by ReptiIe** [ct-js/ct-js#24]
EDIT: 05/12/2012
Seems there are some translations/labels that exist in the file, but are "not connected" (linked) in the main editor, 
or there is some issue with them. Here I will try to update and point out what I will find.

ERROR 1:
Missing translation ID for a tooltip.

New project -> New asset -> Placeholder texture -> Background color (choose color for this window)
![image](https://github.com/user-attachments/assets/1dc5e3c0-b1cd-46cf-bfd2-93290e25a38c)

For the color bar on the right (it has a red border)
There is a tooltip pop up as I notice.

This one does not have a translation ID.
`Please enter a valid value. The two nearest valid values are 224 and 225.` 

It is not a big issue, and critical, but might be worth to fix it one day.
(if I will notice any other missing IDs, I will write it down)

**Comment by ReptiIe** [comment]
ERROR 2: 05/12/2024

Assets -> new asset -> Placeholder texture
Let we create something... doesn't matter what. The default will do just fine. OK -> Save

Next, let we try to edit this placeholder texture:
In the section `Export Settings` there is this check box `Export as blank texture`
next to it there is information icon. It does nothing. 
Here:
![image](https://github.com/user-attachments/assets/4b337c65-2b4c-4d36-9d7c-1759d9930aa0)

It should show tooltip `blankTextureNotice` (line 1612 as I guess)

**Comment by ReptiIe** [comment]
ERROR 3: 08/12/2024

main ct.js menu -> License, scroll down a bit.

Not sure about it, my guess is, that something there went wrong.  Not everything should be in a "h1" correct?
![image](https://github.com/user-attachments/assets/b7adef41-bff0-4877-813a-f99f1255f948)

**Comment by ReptiIe** [comment]
ERROR 4: 08/12/2024

main menu ct.js -> Font for code -> Custom
This window:

![image](https://github.com/user-attachments/assets/24447f04-a22f-4d06-8767-ee07a09be569)

We have [Cancel] and [OK] buttons, but they do not have any "translation ID" for them.


PS: I will be adding here "errors"/"missing translations" etc. once the app get some update/fixes, I will be sure to strike down those errors 😸

**Comment by CosmoMyzrailGorynych** [comment]
Error 1 is shown by the browser (nw.js) itself, though probably a custom title can be shown.
It also should be checked in the neutralino branch, as using the built-in browser most likely will already show the tooltip in OS default language.

**Comment by ReptiIe** [comment]
ERROR 5: 25/12/2024

main ct.js menu -> License, scroll down a bit.

Explaining:
as I see after main CTJS License, there are License for other things, like example Themes.

Usually it has this template:
-> Title
-> Comment (if any)
-> Frame with license

Now I am ok with skipping the "Title" section as well as the "Frame with license" for each theme/app/module/
but I guess it would not hurt to add lines for those "Comments" 

So it would be nice to add:
Lines for comments 

**1) Arkham theme for VSCode**
-> The ct.js'theme called Lucas Dracula is a rough....

**2) Forest Night theme for VSCode**
-> Original theme made by Sainnhe Park; adapted for ct.js by Comigo.

**3) Open Sans font by Steve Matteson**
-> Released under Apache License 2.0 // I guess here "Released under" 

**4) Synthwave '84 theme by robb0wen MIT**
-> This one I believe it has some error. I mention this in "**ERROR 3**" above.

**5) Horizon theme**
-> ct.js has an adapted version of this theme. The original theme...

**6) Nord Theme**
-> ct.js has an adapted version of this theme. The original color...

**7) node-unzipper**
-> Commits in this fork are (c) Ziggy Jonsson...

**8) Rose Pine**
-> ct.js has an adapted version of this theme. The original color...

Example below:
![image](https://github.com/user-attachments/assets/b095e500-4fe7-4e79-ac56-e3c87b55996f)
And as I say - it is not necessary, it just something that tells me it should be translated.

**Comment by CosmoMyzrailGorynych** [comment]
Should any of these license blurbs be? License texts are usually left as is to avoid any translation distortions.

**Comment by ReptiIe** [comment]
I mean only the comments that I marked. Only that. 

Titles and LICENSE itself (in the big frames) I would left as they are.

**Comment by ReptiIe** [comment]
ERROR 6: 29/12/2024

Missing translation for:
Collision with a group / Collision with a template
Group (name) / Template

Here are the screens:

COLLISION WITH A GROUP
![image](https://github.com/user-attachments/assets/1b3c1730-0aab-4338-9dce-2cc6a13f0522)

COLLISION WITH A TEMPLATE
![image](https://github.com/user-attachments/assets/8a237707-d0c2-4e0d-b87a-a66dab5ce945)


I found this in:
-> Make new behavior
-> Add event -> Collision with a group / Collision with a template

**Comment by CosmoMyzrailGorynych** [comment]
Catmods and their events are translated in their module.json

**Comment by ReptiIe** [comment]
Hmm.. sorry for bringing a dumb question - this (marked in red)
this is from.... catmod??

![image](https://github.com/user-attachments/assets/512fc88d-8851-4613-9298-963adc159f9b)

**Comment by CosmoMyzrailGorynych** [comment]
Yes, `place` is a catmod, and these are its events

**Comment by ReptiIe** [comment]
So I understand to change that file - just simply add what is missing...
if I will see example "name_XX" (name_Ru) 
-> Copy, 
-> rename to: name_Pl
-> Add a translation

And that what be it?

If so - if I will finish this one (catmod: place) - make a Pull Request?

EDIT: 
Already done it. I send the PR.

**Comment by ReptiIe** [comment]
ERROR 7: 02/01/2025

After exporting the game to desktops, I received this message:

![image](https://github.com/user-attachments/assets/1ec43336-731e-4458-b1a8-b5faa37a3a6d)


Missing translation.

**Comment by ReptiIe** [comment]
ERROR 8: 13/01/2025

We have this in ENG translation:
`"set maxY": " Set right boundary to",`

This is WRONG, it is no `Set **right** boundary to` but 
`Set **bottom** boundary to` as I guess. 
Will send a PR for this.

**Comment by ReptiIe** [comment]
ERROR 9: 16/01/2025

Missing translations (I believe) 
here is a screen:

![Image](https://github.com/user-attachments/assets/fce9b278-967c-48b3-be1d-46c3aae8690b)

1) missing `with` (like the first one that I marked)
2) missing `replace all by regex`, although I see this one in the file (line 629 - `replaceAllByRegex`)
3) missing `by` (or again it is not linked? Not sure about this one)

Additional question: 
operation blocks like for example this one, orange `(condition)` 
should we leave it and not translate it? not sure if I SHOULD translate things like this. . . There are also other blocks like `var val string value TRUE FALSE` etc. My guess is that this kind of things I should not touch.


And please REOPEN the issue 😄 
I had click it by mistake.

**Comment by CosmoMyzrailGorynych** [comment]
"If (condition)": these are arguments names, and currently there is no mechanism to translate them. (Possible to code tho)

**Comment by ReptiIe** [comment]
ERROR 10 : 26/01/2025

Missing translation for error, screen below

![Image](https://github.com/user-attachments/assets/97ba92c8-93cd-4711-8bd9-8dd36b02e6ac)

You will receive this error note once you will try to drag and drop a file that is not supported.
Would be wise to add a translation string for this one.

It is a reminder.

**Comment by ReptiIe** [comment]
ERROR 11: 30/01/2025

Another missing translations

Here we got an error that is like this:
I had clicked New resources, add fonts.
Added font - it was OK! 

I clicked again the same, add font, AGAIN the same font and I got this:

![Image](https://github.com/user-attachments/assets/cbddf1d1-e5c7-400c-aa4d-05c8b14a85bf)

Perhaps it should show something like "You already have this font" ?


ERROR 12: 30/01/2025
Missing pop up translation (?)

In dev.tools, I guess the last one that is for toogle, it will not display any "pop ups"

![Image](https://github.com/user-attachments/assets/536dfe9d-c663-4301-8128-290eda72d8e3)

**Comment by ReptiIe** [comment]
ERROR 13: 30/01/2025

Missing translation for pop up message, once you will export the game (on windows, export for desktops)

This is what we get (once the export is done and success):

![Image](https://github.com/user-attachments/assets/ced74036-049b-4624-b04e-35e3f1b1a458)

**Comment by ReptiIe** [comment]
ERROR 14: 31/01/2025

Missing translations for `repeat` options in ROOM -> Backgrounds.

I had made a screen in the debugging translation
Here is the part
![Image](https://github.com/user-attachments/assets/caeb8121-b6e5-4cd8-afdd-0b75c1be8cea)
